### PR TITLE
chore: hone e2e test scaffold and run cases in CI

### PR DESCRIPTION
**Pull request by tokers**

### CODE REVIEW

File: .github/workflows/e2e-test-ci.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Install minikube
+        uses: abstractj/setup-minikube@issue-12
+      - name: Output cluster info
+        run: kubectl cluster-info
+      - name: Add images
+        run: |
+          minikube cache add apache/apisix:latest -v 7 --alsologtostderr
+          minikube cache add bitnami/etcd:3.4.14-debian-10-r0 -v 7 --alsologtostderr
+          minikube cache add kennethreitz/httpbin -v 7 --alsologtostderr
+          minikube cache add viewking/apisix-ingress-controller:dev -v 7 --alsologtostderr
```
Comment (gxthrj) [inline .github/workflows/e2e-test-ci.yml:25]: Why adding `viewking/apisix-ingress-controller:dev` cache ?
Do we need to deploy `apisix-ingress-controller`in minikube ?

File: test/e2e/ingress/sanity.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+							Path: "/ip",
+							Backend: scaffold.ApisixRouteRuleHTTPBackend{
+								ServiceName: backendSvc,
+								ServicePort: backendSvcPort[0],
+							},
+						},
+					},
+				},
+			},
+		})
```
Comment (gxthrj) [inline test/e2e/ingress/sanity.go:50]: I think we need to use yaml to manage the creation of `ApisixRoute`, In this way, we can simulate the user's usage scenario test conversion process.

This is an enhancement and will not block PR .

File: test/e2e/testdata/apisix-gw-config-default.yaml [hunk 3]
```diff
@@ -59,6 +59,7 @@ apisix:
 
   allow_admin:                  # http://nginx.org/en/docs/http/ngx_http_access_module.html#allow
     - 127.0.0.0/24              # If we don't set any IP list, then any IP access is allowed by default.
+    - 0.0.0.0/0
```
Comment (gxthrj) [inline test/e2e/testdata/apisix-gw-config-default.yaml:62]: Should remove this.  Had add this in `test/e2e/testdata/apisix-gw-config.yaml` already.

File: Makefile [hunk 4]
```diff
@@ -48,6 +49,14 @@ gofmt:
 unit-test:
 	go test -cover -coverprofile=coverage.txt ./...
 
+### e2e-test:         Run e2e test cases
+e2e-test:
+	export APISIX_ROUTE_DEF=$(PWD)/samples/deploy/crd/v1beta1/ApisixRoute.yaml && \
+	export APISIX_UPSTREAM_DEF=$(PWD)/samples/deploy/crd/v1beta1/ApisixUpstream.yaml && \
+	export APISIX_SERVICE_DEF=$(PWD)/samples/deploy/crd/v1beta1/ApisixService.yaml && \
+	export APISIX_TLS_DEF=$(PWD)/samples/deploy/crd/v1beta1/ApisixTls.yaml && \
+	cd test/e2e && go test github.com/api7/ingress-controller/test/e2e
```
[the remark this answers is not in the document]
Comment (tokers) [reply Makefile:58]: It only calculates the code coverage of package test/e2e, not our ingress controller.

File: .github/workflows/e2e-test-ci.yml (hunk 1 above)
Comment (tokers) [reply .github/workflows/e2e-test-ci.yml:25]: Oops, it seems i made a mistake here, we should build ingress-controller image each time.

File: test/e2e/testdata/apisix-gw-config-default.yaml (hunk 3 above)
Comment (tokers) [reply test/e2e/testdata/apisix-gw-config-default.yaml:62]: OK.

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (gxthrj) [reply Makefile:58]: Yes, we need to calculates code coverage for e2e and UT separately .

[the remark this answers is not in the document]
Comment (tokers) [reply Makefile]: Fixed.

File: .github/workflows/e2e-test-ci.yml (hunk 1 above)
Comment (tokers) [reply .github/workflows/e2e-test-ci.yml]: Fixed.

File: test/e2e/testdata/apisix-gw-config-default.yaml (hunk 3 above)
Comment (tokers) [reply test/e2e/testdata/apisix-gw-config-default.yaml]: Fixed.

**Review by gxthrj — APPROVED** [review]

**Comment by tokers** [comment]
@membphis Please take a look when you have time.

**Review by moonming — APPROVED** [review]

**Pull request merged**